esp-hal/src/spi/master.rs

```diff
@@ -923,8 +932,12 @@ where
     #[instability::unstable]
     pub fn with_sio0(mut self, mosi: impl PeripheralOutput<'d>) -> Self {
         let mosi = mosi.into();
-        mosi.set_output_enable(true);
+
+        mosi.apply_input_config(&InputConfig::default());
+        mosi.apply_output_config(&OutputConfig::default());
+
         mosi.set_input_enable(true);
+        mosi.set_output_enable(false);
```

Comment:
false?

Reply:
Doesn't matter, the peripheral should be in control of the output enable. Even then, initially we wouldn't want to drive a bidirectional line I think.